src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseAsyncMethodInAsyncContext.cs

```diff
@@ -162,29 +163,46 @@ public SyncBlockingSymbol(string Name, string Namespace, SymbolKind Kind, ISymbo
             public ISymbol Value { get; set; }
         }
 
-        private static void GetTypeAndAddToDictionary(string key, string typeName, ConcurrentDictionary<string, INamedTypeSymbol> syncBlockingTypes, Compilation compilation)
+        private static void GetTypeAndAddToDictionary(string key, string typeName, ConcurrentDictionary<string, INamedTypeSymbol> syncBlockingTypes, WellKnownTypeProvider wellKnownTypeProvider)
         {
-            if (compilation.TryGetOrCreateTypeByMetadataName(typeName, out INamedTypeSymbol? typeValue))
+            if (wellKnownTypeProvider.TryGetOrCreateTypeByMetadataName(typeName, out INamedTypeSymbol? typeValue))
             {
                 syncBlockingTypes.AddOrUpdate(key, typeValue, (k, v) => v);
             }
         }
 
-        private static void GetSymbolAndAddToList(string symbolName, string Namespace, SymbolKind kind, List<SyncBlockingSymbol> syncBlockingSymbols, Compilation compilation)
+        private static void GetSymbolAndAddToList(string symbolName, string metadataName, SymbolKind kind, List<SyncBlockingSymbol> syncBlockingSymbols, WellKnownTypeProvider wellKnownTypeProvider)
         {
-            if (compilation.TryGetOrCreateTypeByMetadataName(Namespace, out INamedTypeSymbol? typeValue))
+            if (wellKnownTypeProvider.TryGetOrCreateTypeByMetadataName(metadataName, out INamedTypeSymbol? typeValue))
             {
                 ISymbol? symbolValue = typeValue
                     .GetMembers(symbolName)
                     .FirstOrDefault(s => s.Kind == kind);
 
                 if (symbolValue is not null)
                 {
-                    syncBlockingSymbols.Add(new SyncBlockingSymbol(symbolName, Namespace, kind, symbolValue));
+                    syncBlockingSymbols.Add(new SyncBlockingSymbol(symbolName, metadataName, kind, symbolValue));
                 }
             }
         }
 
+        private static IEnumerable<IMethodSymbol> GetExcludedMethods(WellKnownTypeProvider wellKnownTypeProvider)
+        {
+            var methods = Enumerable.Empty<IMethodSymbol>();
+            if (wellKnownTypeProvider.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftEntityFrameworkCoreDbContext, out INamedTypeSymbol? dbContextType))
+            {
+                methods = methods
+                    .Concat(dbContextType
+                        .GetMembers("Add")
+                        .OfType<IMethodSymbol>())
+                    .Concat(dbContextType
+                        .GetMembers("AddRange")
+                        .OfType<IMethodSymbol>());
+            }
+
+            return methods;
+        }
```

Comment:
Lets use `ImmutableArray<IMethodSymbol>` instead
```suggestion
        private static ImmutableArray<IMethodSymbol> GetExcludedMethods(WellKnownTypeProvider wellKnownTypeProvider)
        {
            var methods = ImmutableArray.CreateBuilder<IMethodSymbol>();
            if (wellKnownTypeProvider.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftEntityFrameworkCoreDbContext, out INamedTypeSymbol? dbContextType))
            {
                foreach (var method in dbContextType.GetMembers().OfType<IMethodSymbol>())
                {
                    if (method.Name is "Add" or "AddRange")
                    {
                        methods.Add(method);
                    }
                }
            }

            return methods.ToImmutable();
        }
```